java/java.source.base/src/org/netbeans/modules/java/source/ElementUtils.java

```diff
@@ -60,6 +60,11 @@ public static TypeElement getTypeElementByBinaryName(JavacTask task, String name
         for (ModuleElement me : allModules) {
             TypeElement found = getTypeElementByBinaryName(task, me, name);
 
+            if (result == found) {
+                // avoid returning null, partial fix for [NETBEANS-4832]
+                continue;
+            }
+
```

Comment:
Not related to this bug, but a few lines below: `(elem.getKind() == ElementKind.CLASS || elem.getKind() == ElementKind.INTERFACE)` IMHO does not work for `ANNOTATION_TYPE` and `ENUM`. Could be worth to fix as
`elem.getKind().isClass() || elem.getKind().isInterface` ?

Reply:
So you're saying that the `if(result==found)continue` seems OK.

And that there's an additional issue that could arise in some cases; and you've proposed a fix which can be incorporated with this change.

Reply:
Added separate commit for ANNOTATION_TYPE, ENUM, still in this PR.

Reply:
@sdedic Known issues handled, please review.